lib/std/os.zig

```diff
@@ -3237,3 +3237,20 @@ pub fn sched_yield() SchedYieldError!void {
         else => return error.SystemCannotYield,
     }
 }
+
+/// Set a socket's options.
+pub fn setsockopt(fd: fd_t, level: u32, optname: u32, opt: []const u8) !void {
+    switch (errno(system.setsockopt(fd, level, optname, opt.ptr, @intCast(socklen_t, opt.len)))) {
```

Comment:
Where did you find out the list of possible errors for this? Here's some that I found:

 * [FreeBSD](https://www.unix.com/man-page/FreeBSD/2/setsockopt/)
 * [Linux](https://linux.die.net/man/2/setsockopt)
 * [OpenBSD](https://man.openbsd.org/setsockopt.2)
 * [Windows](https://docs.microsoft.com/en-us/windows/win32/api/winsock/nf-winsock-setsockopt)

None of these have EDOM or ENOMEM, for example.

Reply:
I got them from [POSIX](https://pubs.opengroup.org/onlinepubs/9699919799/functions/setsockopt.html)..? I'll update it accordingly.

Reply:
Sorry, that's actually my mistake - I didn't find that document, and I think handling all the error codes that POSIX specifies (how you had it before) is wise. I feel bad for messing it up so I'll fix it myself before merging.